validation-test/Sema/type_checker_perf/fast/sr139.swift

```diff
@@ -1,5 +1,5 @@
 // RUN: %target-typecheck-verify-swift
-// RUN: %target-swift-frontend -typecheck %s
+// REQUIRES: tools-release,no_asserts
```

Comment:
Could you mean remove `no_asserts`? It would be great to audit other perf tests for this as well. Ideally we'd like to run perf tests as part of CI which by default runs with assertions on.

Reply:
Let's have a separate discussion about removing `no_asserts` from the perf tests. Either that decision is right for all of them or none of them, but it shouldn't be changed on an ad-hoc basis and the tests shouldn't have inconsistent policy. More so, if running timing sensitive tests as a part of CI is important, then we really ought to lower the default test timeout from 50 minutes to something under 5 seconds. This might require moving a few more tests to `long_test` but I think that's the right tradeoff.

Reply:
I think all of the tests supposed no be checked as part of the normal CI, it was a mistake to separate them into this "no_asserts" category which is completely arbitrary (I can tell you that as a person who have added a majority of these performance tests). What tends to happen is as follows - before committing performance optimizations and/or diagnostic improvements implementers forget (or don't known) to run `no assertions` build which ends up being a problem later and leads to rushed attempts to fix the problem when found. You can find plenty examples of that if you search in closed PRs, here is the most recent one - https://github.com/apple/swift/pull/28116. So marking more performance tests as "no_assert" doesn't really improve situation here. Moreover all of the "fast" tests I think it's worth to have all "fast" tests limited to 1s via `-solver-expression-time-threshold=1` otherwise they should be considered "slow".

Reply:
FWIW – I just tried removing `no_asserts` from all the tests and then I timed a release+asserts run of just the Sema/type_checker_perf tests. It seems like a few of the "fast" tests aren't fast, right? Namely: sr139, rdar23327871, and rdar25866240. Also, it seems like rdar46713933 is missing `-solver-expression-time-threshold=1`, correct? Want a bug report? Want me to remove `no_asserts`?

```
Slowest Tests:
--------------------------------------------------------------------------
2.16s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar33935430.swift
2.16s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar33476240.swift
2.17s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar22949523.swift
2.17s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar30631148.swift
2.18s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar19612086.swift
2.18s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar19915443.swift
2.19s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar18994321.swift
2.19s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar19368383.swift
2.20s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar23224583.swift
2.21s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar22079400.swift
2.21s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar23861629.swift
2.23s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar23682605.swift
2.37s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar54926602.swift
2.52s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar20959612.swift.gyb
3.60s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar21198787.swift.gyb
4.00s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar54580427.swift
5.25s: Swift(linux-x86_64) :: Sema/type_checker_perf/fast/rdar25866240.swift.gyb
5.70s: Swift(linux-x86_64) :: Sema/type_checker_perf/fast/rdar23327871.swift.gyb
16.36s: Swift(linux-x86_64) :: Sema/type_checker_perf/fast/sr139.swift
19.81s: Swift(linux-x86_64) :: Sema/type_checker_perf/slow/rdar46713933.swift

Tests Times:
--------------------------------------------------------------------------
[    Range    ] :: [               Percentage               ] :: [ Count ]
--------------------------------------------------------------------------
[ 0.0s, 1.0s) :: [***********************                 ] :: [ 60/101]
[ 1.0s, 2.0s) :: [******                                  ] :: [ 16/101]
[ 2.0s, 3.0s) :: [*******                                 ] :: [ 19/101]
[ 3.0s, 4.0s) :: [                                        ] :: [  1/101]
[ 4.0s, 5.0s) :: [                                        ] :: [  1/101]
[ 5.0s, 6.0s) :: [                                        ] :: [  2/101]
[ 6.0s, 7.0s) :: [                                        ] :: [  0/101]
[ 7.0s, 8.0s) :: [                                        ] :: [  0/101]
[ 8.0s, 9.0s) :: [                                        ] :: [  0/101]
[ 9.0s,10.0s) :: [                                        ] :: [  0/101]
[10.0s,11.0s) :: [                                        ] :: [  0/101]
[11.0s,12.0s) :: [                                        ] :: [  0/101]
[12.0s,13.0s) :: [                                        ] :: [  0/101]
[13.0s,14.0s) :: [                                        ] :: [  0/101]
[14.0s,15.0s) :: [                                        ] :: [  0/101]
[15.0s,16.0s) :: [                                        ] :: [  0/101]
[16.0s,17.0s) :: [                                        ] :: [  1/101]
[17.0s,18.0s) :: [                                        ] :: [  0/101]
[18.0s,19.0s) :: [                                        ] :: [  0/101]
[19.0s,20.0s) :: [                                        ] :: [  1/101]
```

Reply:
See also: #28149

Reply:
These slow "fast" test do belong in the slow suite, thank you!